### DISCUSSION TITLE
[Errno 11001] getaddrinfo failed

### DISCUSSION THREAD

**Discussion by enrico0506** [soranoo/TradingView-Free-Webhook-Alerts#22] [Q&A]
Hello, I set it up according to the instruction but I received a Here an error has occurred, reason: [Errno 11001] getaddrinfo failed, I would be glad if someone could tell me how to resolve this e<img width="652" alt="Screenshot 2022-12-22 225650" src="https://user-images.githubusercontent.com/83516144/209232939-e6c52879-12b6-41a8-b1ce-24ae3f1e903c.png">or

**Comment by soranoo** [comment]
The error message `[Errno 11001] getaddrinfo failed` usually indicates that there was a problem resolving the hostname or IP address of a remote server (in this case, means the email server). This can happen for a variety of reasons, such as:
* The hostname or IP address is invalid or does not exist.
* The network connection is down or the DNS server is not responding.
* There is a problem with the network configuration or firewall settings.

To troubleshoot this error, you can try the following:

1. Verify that the `imap_server_address` and `imap_server_port` are correct. (both values are located in `config.toml`)
2. Check the network connection and make sure that it is working properly.
3. Try pinging the email server to see if it is reachable. You do it with the command or using the following python script.
4. Check the DNS settings and make sure that the DNS server is configured correctly. (recommended Google DNS `8.8.8.8` and Cloudflare DNS `1.1.1.1`)

sample ping python script:
```
import socket

# Set the IP address and port of the remote server
ip_address = input("Enter the IP address: ")
port = int(input("Enter the port: "))

# Create a TCP/IP socket
s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

# Set a timeout for the socket
s.settimeout(5)

# Try to connect to the remote server
try:
    s.connect((ip_address, port))
    print("Port is open")
except socket.error as e:
    print(f"Connection error: {e}")
    print("Invalid IP address or port ?")
finally:
    # Close the socket
    s.close()
```
